pkg/ottl/expression.go

```diff
@@ -549,6 +550,89 @@ func (g StandardIntLikeGetter[K]) Get(ctx context.Context, tCtx K) (*int64, erro
 	return &result, nil
 }
 
+// ByteSliceLikeGetter is a Getter that returns []byte by converting the underlying value to an []byte if necessary
+type ByteSliceLikeGetter[K any] interface {
+	// Get retrieves []byte value.
+	// The expectation is that the underlying value is converted to []byte if possible.
+	// If the value cannot be converted to []byte, nil and an error are returned.
+	// If the value is nil, nil is returned without an error.
+	Get(ctx context.Context, tCtx K) (byteSlice, error)
+}
+
+type StandardByteSliceLikeGetter[K any] struct {
+	Getter func(ctx context.Context, tCtx K) (any, error)
+}
+
+//nolint:revive
+func (g StandardByteSliceLikeGetter[K]) Get(ctx context.Context, tCtx K) (byteSlice, error) {
```

Comment:
```suggestion
func (g StandardByteSliceLikeGetter[K]) Get(ctx context.Context, tCtx K) ([]byte, error) {
```
The `byteSlice` type is an internal type to the OTTL grammar, we should just use `[]byte` here. Same for the rest of these changes; by the time we've reached this point in parsing we should be working with the `[]byte` type.